tests/fixtures/cassettes/mythicbeasts/IntegrationTests/test_provider_authenticate.yaml

```diff
@@ -0,0 +1,80 @@
+interactions:
+- request:
+    body: grant_type=client_credentials
+    headers:
+      Accept:
+      - '*/*'
+      Accept-Encoding:
+      - gzip, deflate
+      Connection:
+      - keep-alive
+      Content-Length:
+      - '29'
+      Content-Type:
+      - application/x-www-form-urlencoded
+      User-Agent:
+      - python-requests/2.25.1
+    method: POST
+    uri: https://auth.mythic-beasts.com/login
+  response:
+    body:
+      string:
```

Comment:
Hello @lexitus, and thanks a lot for your PR!

The tests are failing because the content of the `response.body.string` parameters in the VCRpy cassettes are objects, when it is supposed to be strings. For the record, VCRpy register any HTTP exchanges during the tests in files, so they can be replayed later without the need to connect to the actual DNS provider API.

How have you registered these cassettes?

Reply:
Thanks for taking a look. :-)

I am trying to follow the docs for recording the cassettes. It seems I was processing the response incorrectly in _filter_response where I was removing the access token, so the response got saved wrong. The output looks more reasonable now.

Reply:
Indeed, now the tests are fixed and are passing.